File: src/_02_StructuralPattern/_02_03_Composite/java/after/Champion.java
package _02_StructuralPattern._02_03_Composite.java.after;

public class Champion implements Component {

  private Bag bag = new Bag();

  @Override
  public int getPrice() {
    return bag.getPrice();
  }

  public Champion add(Component component) {
    bag.add(component);
    return this;
  }
  
}

File: src/_02_StructuralPattern/_02_03_Composite/Champion.js
import Component from "./Component.js";
import Bag from "./Bag.js";

export default class Champion extends Component {
  /** @type { Bag } */
  #bag;

  constructor() {
    super();
    this.#bag = new Bag();
  }

  /**
   * @param { Component } component 
   * @returns { Champion }
   */
  add(component) {
    this.#bag.add(component);
    return this;
  }

  /**
   * @override
   * @returns { number }
   */
  getPrice() {
    return this.#bag.getPrice();
  }
}